plugins/processors/starlark/starlark_test.go

```diff
@@ -0,0 +1,2605 @@
+	m = Metric('cpu')
+	m.fields['time_guest'] = 2.0
+	m.time = 0
+	return m
+`,
+			input: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_idle": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_guest": 2.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+		},
+		{
+			name: "deepcopy",
+			source: `
+def apply(metric):
+	return [metric, deepcopy(metric)]
+`,
+			input: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_idle": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_idle": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_idle": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+		},
+		{
+			name: "set name",
+			source: `
+def apply(metric):
+	metric.name = "howdy"
+	return metric
+`,
+			input: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{"time_idle": 42.0},
+					time.Unix(0, 0),
+				),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("howdy",
+					map[string]string{},
+					map[string]interface{}{"time_idle": 42.0},
+					time.Unix(0, 0),
+				),
+			},
+		},
+		{
+			name: "set name wrong type",
+			source: `
+def apply(metric):
+	metric.name = 42
+	return metric
+`,
+			input: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{"time_idle": 42.0},
+					time.Unix(0, 0),
+				),
+			},
+			expected: []telegraf.Metric{},
+		},
+		{
+			name: "get name",
+			source: `
+def apply(metric):
+	metric.tags['measurement'] = metric.name
+	return metric
+`,
+			input: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{},
+					map[string]interface{}{"time_idle": 42.0},
+					time.Unix(0, 0),
+				),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("cpu",
+					map[string]string{
+						"measurement": "cpu",
+					},
+					map[string]interface{}{"time_idle": 42.0},
+					time.Unix(0, 0),
+				),
+			},
+		},
+		{
+			name: "getattr tags",
+			source: `
+def apply(metric):
+	metric.tags
+	return metric
+`,
+			input: []telegraf.Metric{
+				testutil.MustMetric(
+					"cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_idle": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric(
+					"cpu",
+					map[string]string{},
+					map[string]interface{}{
+						"time_idle": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+		},
+		{
+			name: "setattr tags is not allowed",
```

Comment:
This would be somewhat difficult to arrange for while keeping the expected semantics because `metric.tags` is not a Starlark dictionary, it is a wrapper around the telegraf.Metric.  If we allowed it to be set to a different type we would need to reserve some space to hold onto it on the struct, and then convert it to tags after the starlark script exits.